File: core/src/main/java/fr/erictruong/android/core/fragment/RecyclerFragment.java
package fr.erictruong.android.core.fragment;

import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import fr.erictruong.android.core.R;

public abstract class RecyclerFragment extends Fragment {

    private RecyclerView recyclerView;

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        recyclerView = (RecyclerView) inflater.inflate(R.layout.recycler_view, container, false);
        return recyclerView;
    }

    public RecyclerView getRecyclerView() {
        return recyclerView;
    }
}

File: app/src/main/java/fr/erictruong/material/training/ui/core/fragment/RecyclerFragment.kt
package fr.erictruong.material.training.ui.core.fragment

import android.os.Bundle
import android.support.v4.app.Fragment
import android.support.v7.widget.RecyclerView
import android.view.Gravity
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import android.widget.ViewAnimator
import fr.erictruong.material.training.R
import org.jetbrains.anko.*
import org.jetbrains.anko.recyclerview.v7.recyclerView

open class RecyclerFragment : Fragment() {

    lateinit var viewAnimator: ViewAnimator
    val recyclerView: RecyclerView by lazy { viewAnimator.getChildAt(RecyclerFragmentUI.POSITION_LIST) as RecyclerView }
    val emptyTextView: TextView by lazy { viewAnimator.getChildAt(RecyclerFragmentUI.POSITION_EMPTY) as TextView }
    val loadingTextView: TextView  by lazy { viewAnimator.getChildAt(RecyclerFragmentUI.POSITION_LOADING) as TextView }

    override fun onCreateView(inflater: LayoutInflater?, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        viewAnimator = RecyclerFragmentUI().createView(AnkoContext.create(context, this))
        return viewAnimator
    }
}

class RecyclerFragmentUI : AnkoComponent<Fragment> {

    companion object {
        val POSITION_LIST = 0
        val POSITION_LOADING = 1
        val POSITION_EMPTY = 2
    }

    override fun createView(ui: AnkoContext<Fragment>) = with(ui) {
        viewAnimator {
            recyclerView {
                verticalPadding = dimen(R.dimen.padding_small)
                clipToPadding = false
            }
            textView {
                gravity = Gravity.CENTER
                textResource = R.string.loading
            }
            textView {
                gravity = Gravity.CENTER
                textResource = R.string.empty
            }
        }
    }
}